babel_edtf/__init__.py

```diff
@@ -174,6 +175,23 @@ def _format_edtf0_interval_naive(edtf_interval, format, locale):
         dt_start, dt_end, skeleton, fuzzy=True, locale=locale)
 
 
+def _validate_leap_year(edtf_date):
+    """Validate that dates on the 29th of February are on leap years."""
+    if isinstance(edtf_date, Date):
+        if (
+            edtf_date.precision == PRECISION_DAY
+            and edtf_date.day == '29'
+            and edtf_date.month == '02'
+            and not calendar.isleap(int(edtf_date.year))
+        ):
+            raise EDTFParseException(
+                'Day is out of range for month of February on non-leap year.'
+            )
+    elif isinstance(edtf_date, Interval):
+        _validate_leap_year(edtf_date.lower)
+        _validate_leap_year(edtf_date.upper)
+
```

Comment:
Remark: I do not handle `DateAndTime`, but it's something we do not allow in InvenioRDM.

Reply:
[The `metadata.dates` field ](https://github.com/inveniosoftware/invenio-rdm-records/blob/4c99c74b316be2700dbb4161cdc2f4dd935d83a0/invenio_rdm_records/services/schemas/metadata.py#L370) is the only place that uses [`EDTFDateTimeString`](https://github.com/inveniosoftware/invenio-rdm-records/blob/4c99c74b316be2700dbb4161cdc2f4dd935d83a0/invenio_rdm_records/services/schemas/metadata.py#L261) which [allows `DateAndTime`](https://github.com/inveniosoftware/marshmallow-utils/blob/bb1913c03ede43c7053467d411750c77c6de1e5a/marshmallow_utils/fields/edtfdatestring.py#L24).

We could merge and release this for now, since the field is not visible in the upload form, and only REST API programmatic access would cause it to fail.